File: src/main/java/com/lmp/shared/web/AuthHostResolver.java
package com.lmp.shared.web;

import jakarta.annotation.PostConstruct;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import java.net.URI;

/**
 * Source de vérité de l'URL d'issuer OIDC à partir de {@code app.oauth2.issuer-uri}.
 *
 * <p>Mode single-host (depuis 2026-05) : l'issuer URI est dérivé directement
 * de {@code lmp.site.url} dans {@link com.lmp.shared.config.site.SiteConfigManager}
 * — donc {@code authBaseUrl == siteBaseUrl} en pratique. Cette classe expose
 * une URL absolue stable utilisée par les emails de vérification, reset
 * password, et les redirections OAuth2 côté frontend.</p>
 */
@Component
public class AuthHostResolver {

    private final String issuerUri;
    private String authHost;
    private String authBaseUrl;

    public AuthHostResolver(@Value("${app.oauth2.issuer-uri:}") String issuerUri) {
        this.issuerUri = issuerUri;
    }

    @PostConstruct
    void init() {
        if (issuerUri == null || issuerUri.isBlank()) {
            this.authHost = null;
            this.authBaseUrl = null;
            return;
        }
        try {
            this.authHost = URI.create(issuerUri).getHost();
            this.authBaseUrl = issuerUri.replaceAll("/+$", "");
        } catch (Exception e) {
            this.authHost = null;
            this.authBaseUrl = null;
        }
    }

    /**
     * @return host de l'issuer OIDC (ex. {@code lmp-services.ca}),
     *         ou {@code null} si non résolu.
     */
    public String getAuthHost() {
        return authHost;
    }

    /**
     * @return URL complète de l'issuer (ex. {@code https://lmp-services.ca}),
     *         ou {@code null} si non résolu.
     */
    public String getAuthBaseUrl() {
        return authBaseUrl;
    }
}

File: backend-dotnet/src/Lmp.Infrastructure/SiteConfiguration/AuthHostResolver.cs
using Lmp.Application.SiteConfiguration;

namespace Lmp.Infrastructure.SiteConfiguration;

/// <summary>
/// Resolves the OIDC issuer URL from <c>app.oauth2.issuer-uri</c>. Faithful port
/// of <c>com.lmp.shared.web.AuthHostResolver</c>. In single-host mode the issuer
/// is derived from <c>lmp.site.url</c>, so <c>authBaseUrl == siteBaseUrl</c>.
/// </summary>
public sealed class AuthHostResolver : IAuthHostResolver
{
    private readonly string? _authHost;
    private readonly string? _authBaseUrl;

    public AuthHostResolver(ISiteConfigManager siteConfig)
    {
        var issuerUri = siteConfig.GetOauth2IssuerUri();
        if (string.IsNullOrWhiteSpace(issuerUri))
        {
            return;
        }

        if (Uri.TryCreate(issuerUri, UriKind.Absolute, out var uri))
        {
            _authHost = uri.Host;
            _authBaseUrl = issuerUri.TrimEnd('/');
        }
    }

    public string? GetAuthHost() => _authHost;

    public string? GetAuthBaseUrl() => _authBaseUrl;
}
